src/sql/src/statements.rs

```diff
@@ -389,6 +394,40 @@ pub fn column_def_to_schema(
             .insert(COMMENT_KEY.to_string(), c.to_string());
     }
 
+    if let Some(fulltext_opts) = &column.extensions.fulltext_options {
```

Comment:
What about extracting this block into a function like `with_fulltext_options`?  Make this function smaller.